### ISSUE TITLE
cns service tags are not not being being saved to state when set for a triton_machine resource

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-triton#4]
_This issue was originally opened by @chorrell as hashicorp/terraform#14978. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

There seems to be an issue with setting CNS tags for a triton_machine resource in v0.9.6. I think it worked in a previous version.

Specifically, if you set a cns service tag on a triton_machine (via the [triton](https://www.terraform.io/docs/providers/triton/) provider):

```
  tags = {
    triton.cns.services = "test"
  }
```

The tag state never seems to get populated to the state file (e.g., you always get `"tags.%": "0"`) . A subsequent `terraform plan` will show something like this:

```
~ triton_machine.test
    tags.%:                   "0" => "1"
    tags.triton.cns.services: "" => "test"
```

The cns service tag is added to the actual resource via the Triton API though.

### Terraform Version

```
$ terraform -v
Terraform v0.9.6
```

### Affected Resource(s)

- triton_machine

### Terraform Configuration Files

Here's a simple config to reproduce. It assumes you have set the various `SDC_*` environment variables (like `SDC_ACCOUNT` etc.):

```hcl
provider "triton" {
  alias        = "us-east-1"
  url          =  "https://us-east-1.api.joyent.com"
}

resource "triton_machine" "test" {
  provider         = "triton.us-east-1"
  name             = "test"
  package          = "g4-highcpu-128M"
  image            = "7b5981c4-1889-11e7-b4c5-3f3bdfc9b88b"

  tags = {
    triton.cns.services = "test"
  }
}
```

### Debug Output

Here's the debug output for the initial `terraform plan` then `terraform apply`:

https://gist.github.com/chorrell/1c809cd7e3a2da5903308e39cde93c1c

### Expected Behaviour

The  `triton.cns.services =` value should be added to the state.

### Actual Behaviour

The  `triton.cns.services =` value never gets saved to state (as far as I can tell).

### Steps to Reproduce

1. `terraform plan`
2. `terraform apply`
3. `terraform plan`

**Comment by cheapRoc** [comment]
I believe this also causes timeout issues when re-applying state changes. Since CNS tags aren't persisted they're attempted to re-apply to existing instances. This causes timeout issues for some reason. I've seen this issue on 0.9.8.

**Comment by zero-below** [comment]
As a ghetto workaround, I was able to add ctrl-v characters before the dots in the .tf file. This allowed terraform to manage the entries, and also create proper entries over in joyent. Seems to work for me on Terraform v0.9.11.

Example:
`triton^V.cns^V.services = foo`

(those ctrl-v's were placed, in vi, by doing `ctrl-v ctrl-v`). 

The downside to this method, though, is that terraform tries to update the entries every time it runs.

**Comment by cheapRoc** [comment]
@zero-below I literally just closed my laptop for the evening after working on this fix. Keep an eye out for the PR!

**Comment by zero-below** [comment]
Cool. Seems like it's probably a bug in upstream terraform, doing improper splitting on the .'s in the string, or perhaps the provider isn't quoting it properly when passing it out?

**Comment by cheapRoc** [comment]
It is an upstream issue and, considering CNS is a first class feature of Triton, we'll be supporting it as such with [its own stanza](https://github.com/terraform-providers/terraform-provider-triton/issues/8#issuecomment-316175311).

**Comment by jen20** [comment]
This should have been addressed in #17, where CNS tags become a first-class stanza in configuration of a machine. Thanks for reporting, @chorrell and @zero-below (FWIW the upstream bug regarding `.` in map keys has not been fixed, but is worked around by #17).
